pkg/yang/parse.go

```diff
@@ -321,3 +327,26 @@ func (p *parser) nextStatement() *Statement {
 		return ignoreMe
 	}
 }
+
+// Checks that we have a statement depth of 0. It's an error to exit
+// the parser with a depth of > 0, it means we are missing closing
+// braces. Note: the parser will error out for the case where we
+// start with an unmatched close brace
+//
+// This test is only done if there are no other errors as
+// we may exit early due to those errors -- and therefore there *might*
+// not really be a mismatched brace issue.
+func (p *parser) checkStatementDepth() {
+	// don't check if there are other errors
+	if p.errout.Len() > 0 {
+		return
+	}
+	if p.statementDepth > 0 {
+		plural := ""
+		if p.statementDepth > 1 {
+			plural = "s"
+		}
+		fmt.Fprintf(p.errout, "%s:%d:%d: missing %d closing brace%s\n",
+			p.lex.file, p.lex.line, p.lex.col, p.statementDepth, plural)
+	}
+}
```

Comment:
Hi @wingeng,
I am not a maintainer, but I'd write this piece with

* honouring early return in case there are no depth detected
* keeping happy path aligned on left

```go
func (p *parser) checkStatementDepth() {
	// don't check if there are other errors or no depth misalignment detected
	if p.errout.Len() > 0 || p.statementDepth == 0 {
		return
	}
	plural := ""
	if p.statementDepth > 1 {
		plural = "s"
	}
	fmt.Fprintf(p.errout, "%s:%d:%d: missing %d closing brace%s\n",
		p.lex.file, p.lex.line, p.lex.col, p.statementDepth, plural)
}
```

Reply:
@hellt, sure. Only one minor change to reflect the comment.

```
if p.errout.Len() > 0 || p.statementDepth < 1
```

Theres a case where there is a negative depth when we have too many closing braces .... although, now that I think about it, there should be an errout for that case.  Oh well, belts and suspenders.

Reply:
Since there is already a test for this I think it's fine.